common/src/test/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/TestAuditRecordReadMarshallable.java

```diff
@@ -135,104 +126,221 @@ public void testUnknownType() throws Exception
     }
 
     @Test
-    public void testIllegalClientAddress() throws Exception
+    public void testIllegalClientAddress()
     {
-        givenNextRecordIs((short) 0, "ecaudit-single", 42L, new byte[]{ 1, 2, 3 }, InetAddress.getByName("5.6.7.8").getAddress(), "john", null, Status.ATTEMPT.toString(), "Some operation");
-
-        AuditRecordReadMarshallable readMarshallable = new AuditRecordReadMarshallable();
+        givenNextRecordIs(defaultValues.butWithClientAddress(new byte[]{ 1, 2, 3 }));
 
         assertThatExceptionOfType(IORuntimeException.class)
         .isThrownBy(() -> readMarshallable.readMarshallable(wireInMock))
-        .withMessageContaining("Corrupt client IP address field");
+        .withMessageContaining("Corrupt client_ip field");
     }
 
     @Test
-    public void testNullClientAddress() throws Exception
+    public void testNullClientAddress()
```

Comment:
Doesn't testIllegalClientAddress() and testNullClientAddress() test the same thing?
The exact code is being executed... The only difference is what happens in InetAddress.getByAddress()...
So we're more or less testing that method with these to different test cases.

Same goes for testIllegalCoordinatorAddress() / testNullCoordinatorAddress()

Reply:
True. But, but, but.... It feels right to check both these corner cases.... :-)
Wouldn't this add some value in a future refactoring job? On the other hand, if that's a valid argument, then I should have test cases for all other fields as well where I check for null and illegal data type and so forth...

Reply:
Never mind, removed the null versions.